lang/js/angular.md

```diff
@@ -0,0 +1,204 @@
+---
+
+For this guide adhere to the [Angular 2 Style Guide] or the [Angular 1 Style Guide] unless stated below which supersedes those rules.
+
+Topics for this document are in the same order as they are in the [Angular 1 Style Guide].
+
+
+IIFE
+----
+
+We don't require the use of [IIFE] for every component. We compile all code into one file, remove all the `use strict` lines and put a `use strict` at the top of the minified file when compiled.
+
+We can get away with this because we do not employ the use of global variables within the Angular scope. It is bad practice to use them, and will be avoided when developing an Angular application.
+
+Generate your code without using superfluous [IIFE]s.
+
+    // controller.js
+    "use strict";
+
+    angular.module("module").controller("SessionsController", function ($scope, dependency) {
+        var variable1, variable2;
+
+        function gotoSession() {
+          /* */
+        }
+
+        function refresh() {
+          /* */
+        }
+
+        function search() {
+          /* */
+        }
+
+        $scope.gotoSession = gotoSession;
+        $scope.refresh = refresh;
+        $scope.search = search;
+        variable1 = [];
+        variable2 = 'Sessions';
+    });
+
+If needing an activation function as can be the case when invoking a controller for a directive wrap everything within an `activate` function.
+
+    // controller.js
+    "use strict";
+
+    angular.module("module").controller("SessionsController", function ($scope, dependency) {
+        this.activate = () => {
+            function gotoSession() {
+              /* */
+            }
+
+            function refresh() {
+              /* */
+            }
+
+            function search() {
+              /* */
+            }
+
+            $scope.gotoSession = gotoSession;
+            $scope.refresh = refresh;
+            $scope.search = search;
+            $scope.sessions = [];
+            $scope.title = 'Sessions';
+        };
+    });
+
+
+Controllers
+-----------
+
```

Comment:
You should have at least a line here explaining the topics that are covered under the Controllers heading. One should not have headings bumping into each other.